### PR TITLE
Added stuff for the authorize endpoint

### PR DESCRIPTION
**Pull request by ericanderson91**

### CODE REVIEW

File: src/pingone.ts [hunk 1]
```diff
@@ -61,27 +62,176 @@ class PingOneOidc {
...
+        this.logger.info(PingOneOidc.name, 'options.PkceRequest is true, generating artifacts for request parameters');
+        const pkceArtifacts = await OAuth.generatePkceArtifacts(options, this.logger);
+        url.concat(`&state=${pkceArtifacts.State}&code_challenge=${pkceArtifacts.CodeChallenge}`);
+
+        if (pkceArtifacts.CodeChallengeMethod) {
+          url.concat(`&code_challenge_method=${pkceArtifacts.CodeChallengeMethod}`);
+          this.logger.debug(PingOneOidc.name, 'options.CodeChallengeMethod was applied to url', pkceArtifacts.CodeChallengeMethod);
+        }
+
+        sessionStorage.set('state', pkceArtifacts.State);
```
Comment (jon-oblander) [inline src/pingone.ts:130]: does this need to be setItem?

File: src/types/authz-options.ts [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+  /** Optional - nonce token to prevent replay attacks */
   Nonce?: string;
+
+  /** Optional - defaults to false, For a Proof Key for Code Exchange (PKCE) authorization request,
+   * the /{{envID}}/as/authorize request must include the code_challenge parameter. */
+  PkceRequest?: boolean;
+
+  /** Optional - defaults to empty, The CodeChallengeMethod property is required if the application's
+   * pkceEnforcement property is set to S256_REQUIRED in PingOne. Otherwise, it is optional. */
+  CodeChallengeMethod?: '' | 'S259';
```
[the remark this answers is not in the document]
Comment (ericanderson91) [reply src/types/authz-options.ts]: Oh yep! Good catch

File: src/pingone.ts (hunk 1 above)
Comment (ericanderson91) [reply src/pingone.ts]: Fixed this as well

**Review by jon-oblander — APPROVED** [review]
reviewed.

**Pull request merged**